DevCycle/DVCClient.swift

```diff
@@ -284,13 +298,24 @@ public class DVCClient {
             }
             self.eventQueue.flush(service: service, user: user) { error in
                 callback?(error)
-                if (!self.eventQueue.isEmpty()) {
-                    DispatchQueue.main.asyncAfter(deadline: .now() + self.flushEventsInterval) {
-                        self.flushEvents(callback: nil)
-                    }
-                }
             }
+        } else {
+            callback?(nil)
+        }
+    }
+    
+    public func close(callback: CloseCompletedHandler?) {
```

Comment:
not sure if we even need a `CloseCompletedHandler`, if we don't think the user will ever need a callback for flush events, then they probably don't need to know when it's done after closing

Reply:
the idea is that they need a way to delay any shutdown operation until after the SDK has fully closed and flushed its events. Basically this gives them a "wrap things up" handler